stdeb/downloader.py

```diff
@@ -49,51 +37,20 @@ def find_tar_gz(package_name, pypi_url='https://pypi.org',
                              'releases %r' % (release, all_releases))
         version = release
     else:
-        default_releases = _call(pypi.package_releases, package_name)
-        if len(default_releases) != 1:
-            raise RuntimeError('Expected one and only one release. '
-                               'Non-hidden: %r. All: %r' %
-                               (default_releases, all_releases))
-        default_release = default_releases[0]
+        default_release = pypi_client.release_versions(package_name, current_only=True)
         if verbose >= 2:
             myprint(
-                'found default release: %s' % (', '.join(default_releases),))
+                'found default release: %s' % (', '.join(default_release),))
 
         version = default_release
 
-    urls = _call(pypi.release_urls, package_name, version)
-    for url in urls:
-        if url['packagetype'] == 'sdist':
-            assert url['python_version'] == 'source', \
-                'how can an sdist not be a source?'
-            if url['url'].endswith(('.tar.gz', '.zip')):
-                download_url = url['url']
-                if 'md5_digest' in url:
-                    expected_md5_digest = url['md5_digest']
-                break
+    download_url, expected_md5_digest = pypi_client.download_url(package_name, version)
 
-    if download_url is None:
-        # PyPI doesn't have package. Is download URL provided?
-        result = _call(pypi.release_data, package_name, version)
-        if result['download_url'] != 'UNKNOWN':
-            download_url = result['download_url']
-            # no download URL provided, see if PyPI itself has download
-            urls = _call(pypi.release_urls, result['name'], result['version'])
```

Comment:
This is the behavior for which I could find no sensible replacement in the current PyPI APIs.

Reply:
Yeah, I don't think the concept carries over to the new API.